parsing/src/common.ts

```diff
@@ -65,6 +65,19 @@ function getFriendlyName(name: string): string {
     return name.replace(/\s*\(.*?\)/g, '');
 }
 
+// Example: Build_GeneratorBiomass_Automated_C
+// Change into "generatorbiomas"
+function getPowerProducerBuildingName(className: string): string | null {
+    const match = RegExp(/Build_(\w+)_/).exec(className);
```

Comment:
Does this make sense to you @Maelstromeous, I was initially confused by the 'compiled once' comment - because you definitely don't compile JavaScript, but after some research, while JavaScript doesn't compile in the traditional sense, creating a regular expression once and reusing it avoids the overhead of recreating it multiple times, which can improve performance, especially in loops or frequently called functions.

So this would be the (rough/JS) solution:

```
const regex = /Build_(\w+)_/;

function getPowerProducerBuildingName(className) {
  const match = regex.exec(className);
```

Reply:
In TypeScript it does actually compile, so Copilot is talking bollocks.

Reply:
Also, if it works, it works, premature optimisation is the root of all evil. Parser completes in miliseconds.